### PR TITLE
Async api docs

### PR DESCRIPTION
**Pull request by dnephin**
Step 1 of #87

6672537: addresses a bug I found when I tried to pass an async client to `get_client()`. the sync client only requires a url (it basically passes everything to `requests`), where the async client required a bunch of kwargs.

de81678: Remove wait/cancel from the http_client interface. These are really operations on the eventual objects returned by the client, not the client itself. Changed api-doc loading to support async loading of all linked api documents. This way all the requests are done at once.

6530d67: crochet blew up when it received a unicode url, so I'm handling it here. I think this will probably be fixed when we use sprocket (or we can handle it in sprocket if it's not already).

### CODE REVIEW

File: swaggerpy/async_http_client.py [hunk 1]
```diff
@@ -46,36 +46,21 @@ def start_request(self, request_params):
         """
         # request_params has mandatory: method, url, params, headers
         request_params = {
-            'method': str(request_params['method']),
+            'method': str(request_params.get('method', 'GET')),
```
Comment (prat0318) [inline swaggerpy/async_http_client.py]: it looks unnecessary, as 'method' is a required field of operation. Validation done earlier would throw up.

Comment (dnephin) [inline swaggerpy/async_http_client.py]: We can't assume this is always called from operation.  

The reason for this change is that we're calling it from the Loader, which doesn't necessarily have to pass a method. Even if it does now, it's not required by `HttpClient` so I don't think it should be required here.

File: swaggerpy/async_http_client.py [hunk 2]
```diff
@@ -46,36 +46,21 @@ def start_request(self, request_params):
...
-            'uri': str(request_params['url'] + '?' + urllib_utf8.urlencode(
-                request_params['params'], True))
+            'headers': listify_headers(request_params.get('headers', {})),
+            'uri': '%s?%s' % (
+                request_params['url'],
+                urllib_utf8.urlencode(request_params.get('params', []), True))
         }
 
+        # crochet only supports bytes for the url
+        if isinstance(request_params['uri'], unicode):
```
Comment (prat0318) [inline swaggerpy/async_http_client.py]: shouldn't the check be with `basestring` instead?

Comment (dnephin) [inline swaggerpy/async_http_client.py]: Nope, str is bytes. If it's already str, we don't want to encode it.

File: swaggerpy/response.py [hunk 3]
```diff
@@ -53,7 +52,7 @@ def cancel(self):
         """Try to cancel the API (meaningful for Asynchronous client)
         """
         self._cancelled = True
-        self._http_client.cancel(self._request)
+        self._request.cancel()
```
Comment (prat0318) [inline swaggerpy/response.py]: for synchronous, `SynchronousEventual` should have also defined `cancel()` for the above call?

Comment (dnephin) [inline swaggerpy/response.py]: Yup, I think you're right. I'll add cancel

File: swagger.py.iml [hunk 4]
```diff
@@ -1,19 +0,0 @@
...
-      <excludeFolder url="file://$MODULE_DIR$/swaggerpy.egg-info" />
-    </content>
-    <orderEntry type="jdk" jdkName="Python 2.7.2 virtualenv at ~/virtualenv/swagger" jdkType="Python SDK" />
-    <orderEntry type="sourceFolder" forTests="false" />
-  </component>
-  <component name="PackageRequirementsSettings">
-    <option name="requirementsPath" value="" />
-  </component>
-</module>
-
```
Comment (dnephin) [inline swagger.py.iml]: I got tired of this thing getting in the way of my tab completion.

No one is using this anymore, right?

**Comment by analogue** [comment]
Fix-n-ship!

**Pull request merged**